frontend/src/api/queryBuilder/getAttributeKeys.ts

```diff
@@ -18,20 +18,25 @@ export const getAggregateKeys = async ({
 	dataSource,
 	aggregateAttribute,
 	tagType,
+	isInfraMonitoring,
 }: IGetAttributeKeysPayload): Promise<
 	SuccessResponse<IQueryAutocompleteResponse> | ErrorResponse
 > => {
 	try {
+		const endpoint = isInfraMonitoring
+			? `/hosts/attribute_keys?dataSource=metrics&searchText=${searchText || ''}`
+			: `/autocomplete/attribute_keys?${createQueryParams({
+					aggregateOperator,
```

Comment:
if there is a complete endpoint change including the `baseInstance` as well then it makes sense to create a new API and do not makes changes here.